uefi/src/proto/network/pxe.rs

```diff
@@ -652,17 +651,17 @@ pub enum BootstrapType {
 ///
 /// Corresponds to the `EFI_PXE_BASE_CODE_DISCOVER_INFO` type in the C API.
 #[repr(C)]
-pub struct DiscoverInfo<T: ?Sized> {
+pub struct DiscoverInfo<const N: usize> {
     use_m_cast: bool,
     use_b_cast: bool,
     use_u_cast: bool,
     must_use_list: bool,
     server_m_cast_ip: IpAddress,
     ip_cnt: u16,
-    srv_list: T,
+    srv_list: [Server; N],
```

Comment:
Could we change this to a DST type instead? That's how we've handled similar things elsewhere in this crate, see for example [`PcrEvent`](https://github.com/rust-osdev/uefi-rs/blob/761bed944cfd24516c8a22baf127f64438e2168f/uefi/src/proto/tcg/v1.rs#L101), and how we convert from a thin pointer to the DST type in [`PcrEvent::from_ptr`](https://github.com/rust-osdev/uefi-rs/blob/761bed944cfd24516c8a22baf127f64438e2168f/uefi/src/proto/tcg/v1.rs#L110). That way we don't need to know what `N` is at compile time.

Reply:
Ah sorry I should have provided an example of that. The way to do it without allocation is to provide a byte buffer. Here's a not-yet-merged example for `PcrEvent`: https://github.com/rust-osdev/uefi-rs/pull/631/files#diff-3c3ebe7fc6a06f58897b24d5e0420ad545037b8cb91dae43d5551037122d19cfR136